test/lit.cfg

```diff
@@ -324,13 +324,14 @@ if test_sdk_overlay_dir is not None:
 test_options = os.environ.get('SWIFT_TEST_OPTIONS')
 config.swift_test_options = ' ' + test_options if test_options else ''
 
-clang_module_cache_path = tempfile.mkdtemp(prefix="swift-testsuite-clang-module-cache")
+lit_config.note("Using test results dir: " + config.swift_test_results_dir)
+
+clang_module_cache_path = os.path.join(config.swift_test_results_dir, 'ModuleCache')
```

Comment:
`swift_test_results_dir` is cleaned for every testsuite execution.
https://github.com/rintaro/swift/blob/4a7bae823873ac6c5a42949c2ed099c6779690ac/test/CMakeLists.txt#L275
Isn't that sufficient?